### ISSUE TITLE
Why re-publishing `hwi` crate as `bitcoin_hwi` ? 

### ISSUE THREAD

**Issue by notmandatory** [BP-WG/descriptor-wallet#38]
Hi @dr-orlovsky I see that you have re-published the [`hwi`](https://crates.io/crates/hwi) crate from [bitcoindevkit/rust-hwi](https://github.com/bitcoindevkit/rust-hwi) as a new [`bitcoin_hwi`](https://crates.io/crates/bitcoin_hwi) crate with some small additional commits. Would you be willing to add these changes to the original project by submitting a PR and yank your `bitcoin_hwi` crate so we don't have two nearly identical projects published to crates.io?

**Comment by dr-orlovsky** [comment]
Hi! My plan is to rewrite the crate from scratch over the summer. I started with cloning for now not to wait for the PRs to be merged - I have to do a lot of releases over the summer. 

Feel free to pick up the commits from here for now to the upstream.

**Comment by notmandatory** [comment]
Where can I find your fork ? I don't mind opening PRs to try and upstream your commits if they're generally useful to other users of `rust-hwi`.

**Comment by dr-orlovsky** [comment]
Hm, I am confused, I thought it is here, should be in `hwi` subdirectory of this repo. Ok, searching for it...

**Comment by dr-orlovsky** [comment]
Seems like I changed it somewhere locally and didn't pushed to GitHub :( Now I can't find it even on disk.

Anyway, the change was in adding more derives to a Error type and adding `Copy/Clone` to some of the data types, so it was ultra-trivial.

The purpose of my future crate rewrite will be using `libusb` directly from rust, avoiding interface through command line

**Comment by notmandatory** [comment]
OK for now I'll put in a PR to `rust-hwi` to add the missing derived traits but of course it's up to you if you want to use it in your project or use your own lib. I do think you should provide a link to the source for your current `bitcoind_hwi` crate so that the original [MIT license](https://choosealicense.com/licenses/mit/) copyright requirement is respected.

**Comment by dr-orlovsky** [comment]
Absolutely. I have very tight schedule until the 26th of June (releasing RGB after 3 yrs of work), but after I will download source from the crates.io and re-create the repo with my changes - and will do a PR against the original repo.

**Comment by notmandatory** [comment]
Great, thanks.

**Comment by dr-orlovsky** [comment]
@notmandatory can you please add license information to the `rust-hwi` crate since it currently misses one and it is unclear which conditions for its use and code modification apply

**Comment by dr-orlovsky** [comment]
I found the repo, here it is: https://github.com/mycitadel/bitcoin_hwi
I checked - most of my changes there were already done in the main repo in the last weeks. The rest are related to a dependency I've added (`amplify`) to do `Display` derives for the error types, which probably would not be accepted upstream. Let me know if you need it.

**Comment by notmandatory** [comment]
> @notmandatory can you please add license information to the `rust-hwi` crate since it currently misses one and it is unclear which conditions for its use and code modification apply

Hi sorry for the delay. The `Cargo.toml` for the `rust-hwi` repo specifies the license is MIT. I've created a small PR to make this more explicit, does this help? https://github.com/bitcoindevkit/rust-hwi/pull/38
